src/main/java/org/veupathdb/service/eda/ss/model/varcollection/VarCollection.java

```diff
@@ -107,13 +107,16 @@ public void buildAndValidate(Entity entity) {
       // collect the union of the vocabularies for the collection vocabulary
       if (valueVar.getVocabulary() != null && !valueVar.getVocabulary().isEmpty()) {
         derivedVocabulary.addAll(valueVar.getVocabulary());
-      }
-      else {
+      } else {
         // do not declare a vocabulary unless all member vars have a vocabulary
         if (useVocabulary) // only log once per collection
           LOG.warn("At least one variable in collection " + _properties.id + " does not have a vocabulary, so collection will not either.");
         useVocabulary = false;
       }
+
+      if (!valueVar.getDataShape().isCompatibleWithCollectionShape(_properties.dataShape, derivedVocabulary)) {
```

Comment:
Curious why you put the type check and shape check in different places?

Reply:
I needed to check that it has values first since I need the derived vocabulary in order to check if the collection shape is compatible.

Reply:
Sorry for the late response, I'm just picking this up. I put out this PR kind of rushed when I thought we needed this for release.